tests/fixtures/auth.py

```diff
@@ -35,3 +48,158 @@ def auth_credentials(fs):
     )
 
     return credentials
+
+
+@pytest.fixture
+def basic_client(monkeypatch):
+    """Select basic auth and return a TestClient."""
+    monkeypatch.delenv("RALPH_RUNSERVER_AUTH_OIDC_ISSUER_URI", raising=False)
+    monkeypatch.delenv("RALPH_RUNSERVER_AUTH_OIDC_AUDIENCE", raising=False)
+    reload(conf)
+    app = reload(api).app
+    return TestClient(app)
+
+
+@pytest.fixture
+def oidc_client(monkeypatch):
+    """Select oidc auth and return a TestClient."""
+    monkeypatch.setenv("RALPH_RUNSERVER_AUTH_OIDC_ISSUER_URI", ISSUER_URI)
+    monkeypatch.setenv("RALPH_RUNSERVER_AUTH_OIDC_AUDIENCE", AUDIENCE)
+    reload(conf)
+    reload(api.auth.oidc)
+    reload(api.auth)
+    app = reload(api).app
+    return TestClient(app)
```

Comment:
Another idea would be to use a configurable single fixture for authenticated test client using markers [1]. WDYT?


[1] https://docs.pytest.org/en/7.1.x/how-to/fixtures.html#using-markers-to-pass-data-to-fixtures

Reply:
Pytest markers are mostly used for test configuration (parametrize, skipif, filterwarning etc). It would also mean adding a custom mark related to authentication in the pytest configuration under `setup.cfg`, which I'm not very keen on.

But I like the idea of having a common fixture returning a TestClient, I let you take a look, I instead used the direct test parametrization to change the fixture.